### DISCUSSION TITLE
Defining different relationships between the two same models

### DISCUSSION THREAD

**Discussion by saberhosneydev** [prisma/prisma#16024] [Q&A]
Hello, this is my current model structure that I came up with after a day of search XD. but it feels kinda off from what I'm trying to achieve, which is
Each department can have many staff ( 1:n ) and each department may have a staff appointed as head ( 1:1 )
so at the end I would access it like this
`DepartmentModel.head => StaffModel`
`DepartmentModel.staff => StaffModel[]`
`StaffModel.department => DepartmentModel`

```prisma
model Staff {
    id             String      @id @default(auto()) @map("_id") @db.ObjectId
    firstName      String
    lastName       String
    jobTitle       String
    role           Role
    clearance      Clearance   @default(None)
    file           File?
    DepartmentHead Department? @relation("departmentHead", fields: [departmentId], references: [id])
    departmentId   String?     @unique @db.ObjectId
    Department     Department? @relation("departmentStaff", fields: [departmentId], references: [id])
}
model Department {
    id    String  @id @default(auto()) @map("_id") @db.ObjectId
    name  String
    staff Staff[] @relation("departmentStaff")
    head  Staff?  @relation("departmentHead")
}
```

**Comment by nurul3101** [comment]
Hey @saberhosneydev 👋 
Could you please elaborate on why you feel this is off?
This is a valid schema, are you facing any specific error in your queries?

**Reply by vladi-stevanovic** [reply]
:wave: Hello @saberhosneydev , I’m going through pending GitHub Discussions to check if you need any help. Is this issue still open?

Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
